0000-modules/motivation.md

```diff
@@ -0,0 +1,146 @@
+by users - even if you're working on a small binary project, just to learn, you
+are likely going to want to add a dependency and break your code into two
+files. Doing those two things introduces the need to use  all four keywords
+that are a part of the module system, and to understand what each of them does.
+
+In our proposal, there are three primary keywords: `use`, `pub`, and `export`.
+And in an early project, only `use` and `pub` are going to be necessary. Not
+only that, the `export` keyword is just a visibility, like `pub`, and can be
+viewed as an extension of the pre-existing knowledge about visibility learned
+when your crate gains an external API.
+
+We believe we will simplify the system by requiring users to learn and do less
+stuff. 
+
+We also believe reducing the syntax improves the ergonomics for advanced users
+as well. Many of these declarations tend to feel like 'boilerplate' - you write
+the only thing you could possibly write. You may forget to write a `mod`
+statement, leading to a compiler error (worsening your edit-compile-debug
+cycle). We, the RFC authors, frequently have this experience.
+
+## Path confusion
+
+A problem, recognized by many users, is a confusion that exists about how paths
+work in Rust. Though this has often been glossed as confusion about the
+difference between absolute and relative paths, we don't believe this is true.
+Very many languages use absolute paths for their import statement, but relative
+paths internally to a module, without demonstrating the same confusion. We
+believe the confusion has two causes, both addressed by this RFC.
+
+### Differences between crate root and other modules
+
+In the crate root, absolute and relative paths are the same today. In other
+languages this tends not to be the case. Instead, a common system is to place
+the root of the current package under some namespace; this RFC proposes to do
+just that, using the new `crate` namespace.
+
+This will make absolute paths different from relative paths in every module,
+including the crate root, so that users do not get mislead into believing that
+`use` statements are relative, or that all paths are absolute.
+
+### Multiple keywords import things
+
+Another problem with the current system is that `extern crate` and `mod` bring
+names into scope, but this is not immediately obvious. Users who have not
+grasped this can be confused about how names come into scope, believing that
+dependencies are inherently in scope in the root, for example (though it is
+actually `extern crate` which brings them into scope). Some users have even
+expressed confusion about how it seems to them that `extern crate` and `mod`
+are "just" fancy import statements that they have to use in seemingly arbitrary
+cases, not understanding that they are using them to construct the hierarchy
+used by the `use` statements.
+
+We solve this problem by removing both the `extern crate` and `mod` statements
```

Comment:
`extern crate` will go. `mod` is only if you need to change visibility of a module.